qucs/qucs/schematic_file.cpp

```diff
@@ -343,13 +343,18 @@ int Schematic::saveSymbolJSON()
 }
 
 // -------------------------------------------------------------
-// Returns the number of subcircuit ports.
+// Returns the number of subcircuit ports. In case Qucs cannot save the file it returns a generic error code (-1) or a flag indicating
+// the user wants to save the document into another location
 int Schematic::saveDocument()
 {
   QFile file(DocName);
   if(!file.open(QIODevice::WriteOnly)) {
-    QMessageBox::critical(0, QObject::tr("Error"),
-				QObject::tr("Cannot save document!"));
+    int ret = QMessageBox::critical(this, QObject::tr("Error"),
+                QObject::tr("Cannot save document!\nDo you want to save it into a different folder?"), (QMessageBox::No|QMessageBox::Yes), QMessageBox::Yes);
```

Comment:
*< picky mode on>* I'll say *with a different name*, since it may still be possible to save it in the same folder, but the (full)name must be changed *< /picky mode off>*

Besides, we may avoid asking the user and simply show the MessageBox and calling `SaveAs()` afterwards anyway; `Schematic::saveDocument()` will continue to return `-1` for a failed save and `QucsApp::saveFile()` will take care of calling `SaveAs()`